test/topology_custom/test_topology_recovery_basic.py

```diff
@@ -0,0 +1,80 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+import asyncio
+import logging
+import pytest
+import time
+
+from test.pylib.manager_client import ManagerClient
+from test.pylib.util import wait_for_cql_and_get_hosts
+from test.topology.util import reconnect_driver, restart, enter_recovery_state, \
+        delete_raft_data_and_upgrade_state, log_run_time, wait_until_upgrade_finishes as wait_until_schema_upgrade_finishes, \
+        wait_until_topology_upgrade_finishes, delete_raft_topology_state, check_system_topology_and_cdc_generations_v3_consistency
+
+
+@pytest.mark.asyncio
+@log_run_time
+async def test_topology_recovery_basic(request, manager: ManagerClient):
+    cfg = {'enable_user_defined_functions': False,
+           'experimental_features': ['consistent-topology-changes']}
+    servers = await manager.servers_add(3, config=cfg)
+    cql = manager.cql
+    assert(cql)
+
+    logging.info("Waiting until driver connects to every server")
+    hosts = await wait_for_cql_and_get_hosts(cql, servers, time.time() + 60)
+
+    logging.info(f"Restarting hosts {hosts} in recovery mode")
+    await asyncio.gather(*(enter_recovery_state(cql, h) for h in hosts))
```

Comment:
Now we could do this part using maintenance mode -- no need to write to the scylla_local table to go into recovery, but start with maintenance mode config flag (which automatically triggers raft recovery too). I think that would be a nice practical test for maintenance mode.

(We can switch in a follow-up, and the existing recovery tests too, if it's a hassle)

Reply:
Let's do it in a followup